biz.aQute.bndlib/src/aQute/bnd/build/model/BndEditModel.java

```diff
@@ -677,12 +685,86 @@ public void setIncludeSources(boolean includeSources) {
 	}
 
 	public List<String> getPrivatePackages() {
-		return doGetObject(aQute.bnd.osgi.Constants.PRIVATE_PACKAGE, listConverter);
+		List<String> privatePackagesEntries1 = getEntries(Constants.PRIVATEPACKAGE, listConverter);
+		List<String> privatePackagesEntries2 = getEntries(Constants.PRIVATE_PACKAGE, listConverter);
+
+		Set<String> privatePackages = Stream.concat(privatePackagesEntries1.stream(), privatePackagesEntries2.stream())
+			.collect(toSet());
+		return new ArrayList<>(privatePackages);
+	}
+
+	public void setPrivatePackages(List<String> newPackages) {
+		List<String> privatePackagesEntries1 = getEntries(Constants.PRIVATEPACKAGE, listConverter);
+		List<String> privatePackagesEntries2 = getEntries(Constants.PRIVATE_PACKAGE, listConverter);
+		
+		Set<String> privatePackages = Stream.concat(privatePackagesEntries1.stream(), privatePackagesEntries2.stream())
+			.collect(toSet());
+		
+		List<String> addedEntries = disjunction(newPackages, privatePackages);
+		List<String> removedEntries = disjunction(privatePackages, newPackages);
+
+		privatePackagesEntries1.removeAll(removedEntries);
+		if (privatePackagesEntries1.isEmpty()) {
+			removeEntries(Constants.PRIVATEPACKAGE);
+		} else {
+			setEntries(privatePackagesEntries1, Constants.PRIVATEPACKAGE);
+		}
+
+		privatePackagesEntries2.removeAll(removedEntries);
+		if (privatePackagesEntries2.isEmpty()) {
+			removeEntries(Constants.PRIVATE_PACKAGE);
+		} else {
+			setEntries(privatePackagesEntries2, Constants.PRIVATE_PACKAGE);
+		}
+		
+		if (hasPrivatePackageInstruction()) {
+			privatePackagesEntries1.addAll(addedEntries);
+			setEntries(privatePackagesEntries1, Constants.PRIVATEPACKAGE);
+		} else {
+			privatePackagesEntries2.addAll(addedEntries);
+			setEntries(privatePackagesEntries2, Constants.PRIVATE_PACKAGE);
+		}
 	}
 
-	public void setPrivatePackages(List<? extends String> packages) {
-		List<String> oldPackages = getPrivatePackages();
-		doSetObject(aQute.bnd.osgi.Constants.PRIVATE_PACKAGE, oldPackages, packages, stringListFormatter);
+	private void setEntries(List<? extends String> packages, String key) {
+		List<String> oldPackages = getEntries(key, listConverter);
+		doSetObject(key, oldPackages, packages, stringListFormatter);
+	}
+
+	private void removeEntries(String key) {
+		List<String> oldPackages = getEntries(key, listConverter);
+		doRemoveObject(key, oldPackages, null, stringListFormatter);
+	}
+
+	public void addPrivatePackage(String packageName) {
+		String key = hasPrivatePackageInstruction() ? Constants.PRIVATEPACKAGE : Constants.PRIVATE_PACKAGE;
+		List<String> packages = getEntries(key, listConverter);
+		packages.add(packageName);
+		setEntries(packages, key);
+	}
+
+	public void removePrivatePackage(String packageName) {
```

Comment:
This new method should also be private.

Reply:
Actually, why is this method here? It is not used anywhere in this PR. Just adding it for test case use is not sufficient justification, really.

Reply:
fixed